src/app-nordigen/services/nordigen-service.js

```diff
@@ -15,11 +15,14 @@ import * as nordigenNode from 'nordigen-node';
 import * as uuid from 'uuid';
 import config from '../../load-config.js';
 import jwt from 'jws';
+import { secretsService } from '../../services/secrets-service.js';
 
 const NordigenClient = nordigenNode.default;
 const nordigenClient = new NordigenClient({
-  secretId: config.nordigen?.secretId,
-  secretKey: config.nordigen?.secretKey,
+  secretId:
+    secretsService.get('nordigen_secretId') || config.nordigen?.secretId,
+  secretKey:
+    secretsService.get('nordigen_secretKey') || config.nordigen?.secretKey,
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏
```suggestion
  secretId:
    secretsService.get('nordigen_secretId'),
  secretKey:
    secretsService.get('nordigen_secretKey'),
```

Reply:
Ohhh... one more thing here to check: are we calling an SQL query to get these values even if the user is not using nordigen? Say: if they are making a non-nordigen API call? Does the class still get instatiated and thus makes the SQL calls? If it does - we need to figure out a way to fix it.

Reply:
The secret service does not execute any sql calls until either .get(), .set() or .exists() are executed

Reply:
Right, but you're just building a JSON object here with the secret values. Thus the queries would be executed as long as the whole file gets loaded and executed.

A bit tricky to explain.. But basically there is no difference between these two usages.. in both cases the underlying SQL gets executed:

```
// 1:
const hello = { id: secretSecrvive.get("secret") };

// 2:
const nordigen = new Nordigen({ id: secretSecrvive.get("secret") })
```

Reply:
Ah yes, i could create a Map  to cache values?

something like this?

```js
export const secretsService = () => {
  const _cachedSecrets = new Map();
 
  const get = (name) => {
    return _cachedSecrets.get(name) ?? secretsDb.get(name)?.value ?? null;
  };

  const set = (name, value) => {
    const { changes } = secretsDb.set(name, value);
    if (changes) {
      _cachedSecrets.set(name, value);
    }
    return changes;
  };

  const exists = (name) => {
    return Boolean(get(name));
  };
  
  return { get, set, exists };
};
```

Reply:
What happens in this scenario?

1. customer does not have secrets
2. customer attempt to call nordigen (but this obviously fails)
3. customers sets the new credentials
4. customer attempts to call nordigen again

Does the 2nd nordigen call retrieve the new credentials? Or the cached?

--

Just tested: the nordigen keys are currently retrieved only 1x upon server start. Which means in step (4) the old keys would be retrieved. This means we'll need to instatiate a new nordigen service class each time we call the nordigen `actual-server` APIs. Which means a bit of refactors.. sadly 😢

Reply:
Yeah I guess we would have this issue for any solution we would go for regarding updating keys from frontend.

The only workaround solution I got is to create an onUpdate event that can be listened to in the nordigen class. Then we do some logic there to make sure the values update. 

Then the nordigen class should be able to update its own secrets without reinitialising